src/modules/netinstall/NetInstallPage.cpp

```diff
@@ -0,0 +1,139 @@
+#include "NetInstallPage.h"
+
+#include "./widgets/groupselectionwidget.h"
+#include "ui_page_netinst.h"
+#include "GlobalStorage.h"
+#include "JobQueue.h"
+#include "utils/Logger.h"
+
+#include <QFile>
+#include <QMap>
+#include <QTextStream>
+
+#include <KIO/Job>
+#include <KIO/StoredTransferJob>
+
+#include <QtDebug>
+#include <QtGlobal>
+#include <QWidget>
+
+#include <yaml-cpp/yaml.h>
+
+
+NetInstallPage::NetInstallPage( QWidget* parent )
+    : QWidget( parent )
+    , ui( new Ui::Page_NetInst )
+{
+    ui->setupUi( this );
+}
+
+bool
+NetInstallPage::isReady()
+{
+    // nothing to wait for, the data are immediately ready
+    // if the user does not select any group nothing is installed
+    return true;
+}
+
+void NetInstallPage::ReadGroups( const QByteArray& yamlData )
+{
+    YAML::Node groups = YAML::Load( yamlData.constData() );
+    Q_ASSERT( groups.IsSequence() );
+
+    for ( YAML::const_iterator it = groups.begin(); it != groups.end(); ++it )
```

Comment:
Hint: I'm not sure but you might be able to improve this by using YamlUtils.{h,cpp} in libcalamares.